header/store/store.go

```diff
@@ -296,22 +320,27 @@ func (s *store) flushLoop() {
 	defer close(s.writesDn)
 	for headers := range s.writes {
 		if headers != nil {
-			s.pending = append(s.pending, headers...)
+			// once headers are in pending and accessible
+			s.pending.Append(headers...)
+			// notify waiters if any + increase current height
+			// it is important to do Pub after updating pending
+			// so pending is consistent with atomic Height counter on the heightSub
+			s.heightSub.Pub(headers...)
 		}
 		// ensure we flush only if pending is grown enough, or we are stopping(headers == nil)
-		if len(s.pending) < DefaultWriteBatchSize && headers != nil {
+		if s.pending.Len() < DefaultWriteBatchSize && headers != nil {
```

Comment:
This `&&` doesn't match the comment's `or`.

Reply:
It matches because comments mentions `headers == nil` unlike in the code `headers != nil`. But this comment needs to be updated to match those for the sake of readability. These aren't aligned, because I wrote the code with a comment, then updated the code without updating the comment, iirc.